spec/models/state_benefit_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe StateBenefit do
+  let(:gross_income_summary) { create :gross_income_summary }
+
+  let!(:excluded_state_benefit_type) { create :state_benefit_type, exclude_from_gross_income: true }
+
+  describe '.generate!' do
+    context 'an excluded state benefit' do
+      subject { StateBenefit.generate!(gross_income_summary, excluded_state_benefit_type.name) }
+
+      it 'has a link to gross income summary' do
+        expect(subject.gross_income_summary_id).to eq gross_income_summary.id
+      end
+      it 'has a link to state benefit type' do
+        expect(subject.state_benefit_type_id).to eq excluded_state_benefit_type.id
+      end
+      it 'name is blank' do
+        expect(subject.name).to be_blank
+      end
+    end
+    context 'other state benefit' do
+      before do
+        StateBenefitType.find_or_create_by(label: 'other', name: 'Other state benefit type', exclude_from_gross_income: false)
+      end
```

Comment:
I used StateBenefitType.find_or_create_by in the StateBenefit model test
to add the state benefit type 'other' into state benefit types table, if it does not exist.
Cases when 'other' doesn't exist is when the db/seed isn't performed in testing, however, in circleci the seeds are run. There'll be a duplicate 'other' record which will cause an error as there is a uniqueness constraint on label in the state benefit types table.

Reply:
`find_or_create` in tests is a bit of a code smell - it means that you're never sure what state your DB is in when you run the tests.

I think we're better off writing  a factory to create this record and using it in a `let!` statement instead of the one we have in line 6

Reply:
Thanks, I removed this. It was a hack around the test database being seeded in circleci. As we were using db:setup (which seeded the test database) so I changed it to 'db:create db:schema:load' in the circleci config file.